### PR TITLE
[openweathermap] Code clean-up

### PR DESCRIPTION
**Pull request by florian-h05**
- Remove the "Current UV Index and Forecast" Thing as the respective API has been shut down on 1st April 2021. The functionality has been replaced by the One Call API.
- Improve class JavaDoc of DTOs.
- Remove not-needed setters from DTOs. (All required upstream data is sent to OWM by using query params.)

### CODE REVIEW

File: bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml [hunk 1]
```diff
@@ -317,7 +299,7 @@
 			<channel id="icon-id" typeId="condition-icon-id"/>
 			<channel id="temperature" typeId="system.outdoor-temperature"/>
 			<channel id="apparent-temperature" typeId="apparent-temperature"/>
-			<channel id="pressure" typeId="system.barometric-pressure"/>
+			<channel id="pressure" typeId="systembarometric-pressure"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml:302]: This change looks unintended?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the cleanup! Only one finding.

File: bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml (hunk 1 above)
Comment (florian-h05) [reply bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml]: Good catch, thanks!
I've reverted that change in https://github.com/openhab/openhab-addons/pull/16369/commits/1d57122ed94740d21e0075faf3bed9e400ab6e38.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**